models/yang/sonic/sonic-bgp-neighbor.yang

```diff
@@ -48,8 +59,21 @@ module sonic-bgp-neighbor {
             list BGP_NEIGHBOR_AF_LIST {
                 key "vrf_name neighbor_ip afi_safi";
 
+                leaf vrf_name {
+                    type leafref {
+                        path "/sbgpg:sonic-bgp-global/sbgpg:BGP_GLOBALS/sbgpg:BGP_GLOBALS_LIST/sbgpg:vrf";
+                    }
+                   description "Network-instance/VRF name";
+                }
+
                 leaf neighbor_ip {
-                    type inet:ip-address;
+                    type leafref {
+                        path "../../../BGP_NEIGHBOR/BGP_NEIGHBOR_LIST/neighbor_ip";
+                    }
+
+                    must "/sbnbr:sonic-bgp-neighbor/sbnbr:BGP_NEIGHBOR/sbnbr:BGP_NEIGHBOR_LIST[vrf_name=current()/../vrf_name][neighbor_ip=current()]" {
+                            error-message "BGP Neighbor Table is not configured with vrf_name and neighbor_ip";
```

Comment:
Other sonic YANGS gas error like below, may be we can follow the same
error-message "Invalid Instance";
error-app-tag instance-number-invalid;